srv/salt/_modules/cephdisks.py

```diff
@@ -51,23 +62,56 @@ def osd_list(self) -> list:
                     osd_ids.append(osd_id)
         return osd_ids
 
+    @staticmethod
+    def _is_root_disk(path: str) -> bool:
+        """ Return True/False if disk is root disk """
+        rc_, stdout, _ = __salt__['helper.run'](
+            "mount|grep ' / '|cut -d' ' -f 1 | sed 's/[0-9]//g'")
+        if not rc_ == 0:
+            log.warning("Could not determine root disk. Command failed")
+            return False
+        return stdout == path
+
```

Comment:
Not that I've seen anyone use an NVMe for the root device yet, I can imagine it will be coming.  The sed strip would leave 'nvmenp'.

```
        with open("/proc/mounts", "r") as mounts:
            for line in mounts:
                entry = line.split()
                if entry[1] == "/":
                    return entry[0].startswith(path)
```
Not tested, but I think the startswith simplifies worrying about the number partition and the NVMe `p` with number partition.  And I just realized this will return incorrectly for the `/dev/sda` and `/dev/sdaa`.  We may be stuck with a regex and separate NVMe check.

Reply:
Even though the `startswith` thing has problems with `/dev/sda` and `/dev/sdaa`, I have to say I like reading /proc/mounts natively in python, rather than calling out to mount, grep and sed.

Reply:
I switched to `/proc/mount` reading and changed the logic a bit. It's not first detecting the root_disk and saves it. This saves us a couple of iterations through the disks array.